spec/std/option_parser_spec.cr

```diff
@@ -85,6 +85,105 @@ describe "OptionParser" do
     expect_capture_option ["-f", "-g -h"], "-f FLAG", "-g -h"
   end
 
+  describe "Consumption of flags following an ungiven optional argument" do
+    context "Given a short option with an optional value" do
+      it "doesn't eat a following short option" do
+        flag = nil
+        not_eaten = [] of String
+        args = ["-f", "-g", "-i"]
+        OptionParser.parse(args) do |opts|
+          opts.on("-f [FLAG]", "some flag") do |flag_value|
+            flag = flag_value
+          end
+          opts.on("-g", "shouldn't be eaten") do
+            not_eaten << "-g"
+          end
+          opts.on("-i", "shouldn't be eaten") do
+            not_eaten << "-i"
+          end
+        end
+        flag.should eq("")
+        not_eaten.should eq(["-g", "-i"])
+        args.size.should eq(0)
+      end
+      it "doesn't eat a following long option" do
```

Comment:
Please, add whitespaces around `it` and `context` blocks:

```suggestion

      it "doesn't eat a following long option" do
```

ditto elsewhere

Reply:
I ran this through the formatter but it didn't pick them up, I've fixed it now.